src/PythonProjectBootstrapper/ProjectGenerationUtils.py

```diff
@@ -79,31 +82,52 @@ def ConditionallyRemoveUnchangedTemplateFiles(
             original_hash = existing_manifest_dict[removed_file_rel_path]
 
             if current_hash == original_hash:
+                deleted_files.add(removed_full_path.as_posix())
                 removed_full_path.unlink()
 
+    return deleted_files
+
 
 # ----------------------------------------------------------------------
 def CopyToOutputDir(
     src_dir: Path,
     dest_dir: Path,
-) -> None:
+) -> list[set[str]]:
```

Comment:
I'd recommend creating a `dataclass`\-based object as a return value that explicitly defines members for each type of file. `list` implies that there can be 0 or more `set`s. `tuple` would be a better choice (as there will always be 4 sets), but still isn't ideal as the order of sets isn't clear to the caller from looking at the function signature.

```
@dataclass(frozen=True)
class CopyToOutputDirResult:
    deleted_files: set[str]
    added_files: set[str]
    overwritten_files: set[str]
    modified_template_files: set[str]
```